docs/current_docs/tensors.md

```diff
@@ -142,6 +142,131 @@ GaussianVertex vertex = new GaussianVertex(shape, mu, 0);
  */
 ```  
 
+## Tensor Broadcasting
+
+Broadcasting enables you to perform operations across tensors of different shape, rank and length.
+
+* length (the total number of elements)
+* rank (the number of dimensions)
+* shape (the length in each dimension). 
+
+Tensors can be added, subtracted, multiplied and more with each other. 
+It wouldn’t be very useful if you could only operate on tensors of the same shape, fortunately you can perform broadcasting.  
+
+### Tensor and Scalar
+
+Let’s start with a simple example, multiplying each element inside a matrix by a constant.
+
+Let’s define A as a tensor of shape [2, 2]. It’s therefore a 2x2 matrix of rank 2. 
+Let’s define B as a tensor of shape []. It’s therefore a rank 0 constant (a scalar).
+
+```java
+DoubleTensor A = DoubleTensor.create(new double[]{
+    10., 20.,
+    30., 40.
+}, 2, 2); //shape [2, 2]
+
+DoubleTensor B = DoubleTensor.scalar(5.); //shape []
+
+DoubleTensor C = A.times(B); //shape [2, 2]
+```
+
+We can do operations as normal between the two, even though they have different shapes. 
+In the case of a tensor and a scalar, the resulting tensor will always have the shape of the initial tensor.
```

Comment:
I think you can drop "In the case of a tensor and a scalar" - it aids the clarity and we know you are only talking about this case.